dev/release-build.sh

```diff
@@ -0,0 +1,268 @@
+  --releaseVersion - Release identifier used when publishing
+  --releaseRc - Release RC identifier used when publishing
+  --developmentVersion - Release identifier in top level package directory
+  --tag - Release Tag identifier used when taging git
+
+All other inputs are environment variables
+
+GPG_PASSPHRASE - Passphrase for GPG key
+
+Note: Ensure you have set up standard Apache committer settings in ~/.m2/settings.xml as per
+
+   http://maven.apache.org/developers/committer-settings.html
+
+EOF
+  exit 1
+}
+
+set -e
+
+if [ $# -eq 0 ]; then
+  exit_with_usage
+fi
+
+
+# Process each provided argument configuration
+while [ "${1+defined}" ]; do
+  IFS="=" read -ra PARTS <<< "$1"
+  case "${PARTS[0]}" in
+    --release-prepare)
+      RELEASE_PREPARE=true
+      shift
+      ;;
+    --release-publish)
+      RELEASE_PUBLISH=true
+      shift
+      ;;
+    --release-snapshot)
+      RELEASE_SNAPSHOT=true
+      shift
+      ;;
+    --gitCommitHash)
+      GIT_REF="${PARTS[1]}"
+      shift
+      ;;
+    --releaseVersion)
+      RELEASE_VERSION="${PARTS[1]}"
+      shift
+      ;;
+    --developmentVersion)
+      DEVELOPMENT_VERSION="${PARTS[1]}"
+      shift
+      ;;
+    --releaseRc)
+      RELEASE_RC="${PARTS[1]}"
+      shift
+      ;;
+    --tag)
+      RELEASE_TAG="${PARTS[1]}"
+      shift
+      ;;
+    *help* | -h)
+      exit_with_usage
+     exit 0
+     ;;
+    -*)
+     echo "Error: Unknown option: $1" >&2
+     exit 1
+     ;;
+    *)  # No more options
+     break
+     ;;
+  esac
+done
+
+
+for env in GPG_PASSPHRASE; do
+  if [ -z "${!env}" ]; then
+    echo "ERROR: $env must be set to run this script"
+    exit_with_usage
+  fi
+done
+
+if [[ "$RELEASE_PREPARE" == "true" && -z "$RELEASE_VERSION" ]]; then
+    echo "ERROR: --releaseVersion must be passed as an argument to run this script"
+    exit_with_usage
+fi
+
+if [[ "$RELEASE_PREPARE" == "true" && -z "$DEVELOPMENT_VERSION" ]]; then
+    echo "ERROR: --developmentVersion must be passed as an argument to run this script"
+    exit_with_usage
+fi
+
+# Explicitly set locale in order to make `sort` output consistent across machines.
+# See https://stackoverflow.com/questions/28881 for more details.
+export LC_ALL=C
+
+# Commit ref to checkout when building
+GIT_REF=${GIT_REF:-master}
+
+GPG="gpg --no-tty --batch"
+
+BASE_DIR=$(pwd)
+
+MVN="mvn"
+PUBLISH_PROFILES="-Pdistribution"
+
+if [ -z "$RELEASE_RC" ]; then
+  RELEASE_RC="rc1"
+fi
+
+if [ -z "$RELEASE_TAG" ]; then
+  RELEASE_TAG="$RELEASE_VERSION-$RELEASE_RC"
+fi
+
+DEST_DIR_NAME="bahir-$DEVELOPMENT_VERSION"
+
+RELEASE_STAGING_LOCATION="https://dist.apache.org/repos/dist/dev/bahir/"
+
+
+echo "  "
+echo "-------------------------------------------------------------"
+echo "------- Release preparation with the following parameters ---"
+echo "-------------------------------------------------------------"
+echo "Goal                ==> $GOAL"
```

Comment:
GOAL isn't set in this script. If it is assumed to be present in the script's environment, I would suggest documenting it in the beginning.